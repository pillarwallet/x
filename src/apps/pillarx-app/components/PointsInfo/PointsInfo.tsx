// components
import Body from '../Typography/Body';

type PointsInfoProps = {
  icon: string;
  value: React.ReactNode;
  label: string;
  beforeValue?: string;
  afterValue?: string;
  isWhite?: boolean;
};

const PointsInfo = ({
  icon,
  value,
  label,
  beforeValue,
  afterValue,
  isWhite = false,
}: PointsInfoProps) => (
  <div className="flex gap-4 p-3 items-center">
    <img
      src={icon}
      alt={label}
      className={`flex w-[52px] h-[52px] items-center justify-center p-3 rounded-md ${isWhite ? 'bg-white' : 'bg-container_grey'}`}
    />
    <div className="flex flex-col">
      <div className="desktop:text-[22px] tablet:text-lg mobile:text-lg text-white">
        {beforeValue && (
          <span className="desktop:text-base tablet:text-sm mobile:text-sm">
            {beforeValue}{' '}
          </span>
        )}
        {value}
        {afterValue && (
          <span className="desktop:text-base tablet:text-sm mobile:text-sm">
            {' '}
            {afterValue}
          </span>
        )}
      </div>
      <Body className="desktop:text-base tablet:text-sm mobile:text-sm">
        {label}
      </Body>
    </div>
  </div>
);

export default PointsInfo;
