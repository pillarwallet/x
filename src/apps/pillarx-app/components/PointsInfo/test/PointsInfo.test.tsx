import { render, screen } from '@testing-library/react';

// components
import PointsInfo from '../PointsInfo';

const mockIcon = 'https://example.com/icon.png';
const mockValue = '500';
const mockLabel = 'My PX Points';
const mockBeforeValue = '#';
const mockAfterValue = 'PX';

describe('<PointsInfo />', () => {
  it('renders correctly and matches snapshot', () => {
    const tree = render(
      <PointsInfo
        icon={mockIcon}
        value={mockValue}
        label={mockLabel}
        beforeValue={mockBeforeValue}
        afterValue={mockAfterValue}
      />
    );
    expect(tree).toMatchSnapshot();
  });

  it('renders with all the data', () => {
    render(
      <PointsInfo
        icon={mockIcon}
        value={mockValue}
        label={mockLabel}
        beforeValue={mockBeforeValue}
        afterValue={mockAfterValue}
      />
    );

    const icon = screen.getByRole('img');
    expect(icon).toHaveAttribute('src', mockIcon);
    expect(icon).toBeInTheDocument();

    expect(screen.getByText(`${mockBeforeValue}`)).toBeInTheDocument();
    expect(screen.getByText(`${mockValue}`)).toBeInTheDocument();
    expect(screen.getByText(`${mockAfterValue}`)).toBeInTheDocument();

    expect(screen.getByText(mockLabel)).toBeInTheDocument();
  });

  it('renders correctly without beforeValue and afterValue', () => {
    render(<PointsInfo icon={mockIcon} value={mockValue} label={mockLabel} />);

    expect(screen.queryByText(`${mockBeforeValue}`)).not.toBeInTheDocument();
    expect(screen.getByText(`${mockValue}`)).toBeInTheDocument();
    expect(screen.queryByText(`${mockAfterValue}`)).not.toBeInTheDocument();
  });

  it('applies the correct styling when isWhite is true', () => {
    render(
      <PointsInfo icon={mockIcon} value={mockValue} label={mockLabel} isWhite />
    );

    const icon = screen.getByRole('img');
    expect(icon).toHaveClass('bg-white');
  });

  it('applies the default styling when isWhite is false', () => {
    render(<PointsInfo icon={mockIcon} value={mockValue} label={mockLabel} />);

    const icon = screen.getByRole('img');
    expect(icon).toHaveClass('bg-container_grey');
  });
});
