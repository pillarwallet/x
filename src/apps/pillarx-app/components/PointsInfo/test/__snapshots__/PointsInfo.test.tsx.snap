// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<PointsInfo /> > renders correctly and matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex gap-4 p-3 items-center"
      >
        <img
          alt="My PX Points"
          class="flex w-[52px] h-[52px] items-center justify-center p-3 rounded-md bg-container_grey"
          src="https://example.com/icon.png"
        />
        <div
          class="flex flex-col"
        >
          <div
            class="desktop:text-[22px] tablet:text-lg mobile:text-lg text-white"
          >
            <span
              class="desktop:text-base tablet:text-sm mobile:text-sm"
            >
              #
               
            </span>
            500
            <span
              class="desktop:text-base tablet:text-sm mobile:text-sm"
            >
               
              PX
            </span>
          </div>
          <p
            class="text-base font-medium desktop:text-base tablet:text-sm mobile:text-sm"
          >
            My PX Points
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex gap-4 p-3 items-center"
    >
      <img
        alt="My PX Points"
        class="flex w-[52px] h-[52px] items-center justify-center p-3 rounded-md bg-container_grey"
        src="https://example.com/icon.png"
      />
      <div
        class="flex flex-col"
      >
        <div
          class="desktop:text-[22px] tablet:text-lg mobile:text-lg text-white"
        >
          <span
            class="desktop:text-base tablet:text-sm mobile:text-sm"
          >
            #
             
          </span>
          500
          <span
            class="desktop:text-base tablet:text-sm mobile:text-sm"
          >
             
            PX
          </span>
        </div>
        <p
          class="text-base font-medium desktop:text-base tablet:text-sm mobile:text-sm"
        >
          My PX Points
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
